awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -588,3 +589,26 @@ def is_job_done(self, job_status):
             return False
         else:
             return True
+
+    def wait_on_url(self, object_name=None, object_type=None, url=None, timeout=None, interval=None):
+        # Grab our start time to compare against for the timeout
+        start = time.time()
+        result = self.get_endpoint(url)
+        while not result['json']['finished']:
```

Comment:
All the endpoints I checked had finished because they are all job based, but I added in a check that will fail with a pretty message if this is not the case.